### PR TITLE
Bitwise elementwise functions

### PR DESCRIPTION
**Pull request by oleksandr-pavlyk**
Implementation of `dpt.bitwise_invert`, `dpt.bitwise_and`, `dpt.bitwise_or`, `dpt.bitwise_xor`, `dpt.bitwise_left_shift`, and `dpt.bitwise_right_shift` per array API.

- [x] Have you provided a meaningful PR description?
- [x] Have you added a test, reproducer or referred to an issue with a reproducer?
- [x] Have you tested your changes locally for CPU and GPU devices?
- [x] Have you made sure that new changes do not introduce compiler warnings?
- [ ] Have you checked performance impact of proposed changes?
- [ ] If this PR is a work in progress, are you opening the PR as a draft?

### CODE REVIEW

File: dpctl/tensor/libtensor/include/kernels/elementwise_functions/bitwise_left_shift.hpp [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+namespace td_ns = dpctl::tensor::type_dispatch;
+namespace tu_ns = dpctl::tensor::type_utils;
+
+template <typename argT1, typename argT2, typename resT>
+struct BitwiseLeftShiftFunctor
+{
+    static_assert(std::is_integral_v<argT1>);
+    static_assert(std::is_integral_v<argT2>);
+    static_assert(!std::is_same_v<argT1, bool>);
+    static_assert(!std::is_same_v<argT2, bool>);
```
Comment (vtavana) [inline dpctl/tensor/libtensor/include/kernels/elementwise_functions/bitwise_left_shift.hpp:58]: Did you intend to use `is_integral_not_boolean` that you added to `common.hpp` here?

Comment (oleksandr-pavlyk) [reply dpctl/tensor/libtensor/include/kernels/elementwise_functions/bitwise_left_shift.hpp:58]: Thanks for bringing this up. I added that predicate to build a dense inputs integral dtype matrix, but had to give up that idea. It was convenient is the output dtype was always the same the dtype of the first input, but array API mandates adherence to type promotion rules. I should remove the definition of `is_integral_not_boolean` from "common.hpp".

Comment (oleksandr-pavlyk) [reply dpctl/tensor/libtensor/include/kernels/elementwise_functions/bitwise_left_shift.hpp:58]: I have removed definition of `is_integral_not_boolean` from "common.hpp" now.

**Comment by ndgrigorian** [comment]
@oleksandr-pavlyk 
Array API tests have caught a bug:
```
In [1]: import dpctl.tensor as dpt, numpy as np

In [2]: x = dpt.asarray([1, 1])

In [3]: y = dpt.asarray(64)

In [4]: dpt.bitwise_right_shift(x, y)
Out[4]: usm_ndarray([1, 1])

In [5]: x_np, y_np = dpt.asnumpy(x), dpt.asnumpy(y)

In [6]: np.right_shift(x_np, y_np)
Out[6]: array([0, 0])
```

**Comment by ndgrigorian** [comment]
> @oleksandr-pavlyk Array API tests have caught some undesired behavior:
> 
> ```
> In [1]: import dpctl.tensor as dpt, numpy as np
> 
> In [2]: x = dpt.asarray([1, 1])
> 
> In [3]: y = dpt.asarray(64)
> 
> In [4]: dpt.bitwise_right_shift(x, y)
> Out[4]: usm_ndarray([1, 1])
> 
> In [5]: x_np, y_np = dpt.asnumpy(x), dpt.asnumpy(y)
> 
> In [6]: np.right_shift(x_np, y_np)
> Out[6]: array([0, 0])
> ```

And oddly enough:
```
In [8]: dpt.bitwise_right_shift(dpt.asarray([1, 1]), dpt.asarray(62))
Out[8]: usm_ndarray([0, 0])

In [9]: dpt.bitwise_right_shift(dpt.asarray([1, 1]), dpt.asarray(65))
Out[9]: usm_ndarray([0, 0])
```
which is the expected result.

**Comment by oleksandr-pavlyk** [comment]
@ndgrigorian C/C++ spec states that bitshift [behavior is undefined](https://en.wikipedia.org/wiki/Bitwise_operations_in_C) if the second argument is negative or not less than total type bitwidth. Additional branching was added ensure compliance with spec.

**Review by ndgrigorian — APPROVED** [review]
Test is fixed so LGTM!

**Pull request merged**